src/Controller/LtiController.php

```diff
@@ -239,6 +243,14 @@ protected function saveTokenToSession($token)
                 }
             }
 
+            // Context is optional and contains a unique id, label, title, and type
+            if (!empty($token->{'https://purl.imsglobal.org/spec/lti/claim/context'})) {
+                $contextFields = (array) $token->{'https://purl.imsglobal.org/spec/lti/claim/context'};
+                foreach ($contextFields as $key => $val) {
+                    $this->session->set('lms_course_' . $key, $val);
+                }
+            }
+
```

Comment:
The relocation of this code from `src/Lms/D2l/D2lLms.php` seems to have broken UDOIT for us. Any reason why this was moved?

Reply:
@ottenhoff

Reply:
Strangely, it causes our Canvas instance to get an API error

Reply:
I'm going to revert and reopen this until I resolve the issue

Reply:
Maybe you're getting LMS course information from two places: the LTI way and the custom-Canvas-API way?

If the goal is to get this project to be less-custom-Canvas and more-LTI-general, then trying to do things the proper-LTI way makes sense. The claim context is the standard way to get info on the launching course

Reply:
I 100% agree on this. I'll do some digging tomorrow and see if I can't figure it out. UCF uses canvas so the majority of our testing is there, but the standard protocols should be properly supported.